### PR TITLE
externals: Link SDL2 statically.

### PR DESCRIPTION
**Pull request by Steveice10**
* Updates SDL2 and libusb to resolve conflicting GUID definitions on Windows.
  * SDL2 points to the latest stable release, 2.26.5
  * libusb points to the latest API revision number bump.
* Switches included SDL2 over to static linking same as other included externals, since this works now with the aforementioned updates.

Part of a multi-PR effort to update and simplify our external dependencies and release bundling.

- - -
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/6461)

### CODE REVIEW

**Comment by vitor-k** [comment]
Updating SDL will break some [parts](https://github.com/citra-emu/citra/blob/master/src/input_common/sdl/sdl_impl.cpp#L402) of the auto mapping code, so we should either change the struct [here](https://github.com/citra-emu/citra/blob/master/src/input_common/sdl/sdl_impl.cpp#L60) depending on the SDL version or just remove the code altogether (it's a bit hacky and makes little difference)

**Comment by Steveice10** [comment]
Thanks, marking as draft while I take a look.

**Comment by Steveice10** [comment]
Not exactly sure what the implications of removing the workaround are so for now I've just updated the structures to be correct on SDL >= 2.26.0.

File: src/input_common/sdl/sdl_impl.cpp [hunk 1]
```diff
@@ -57,11 +57,35 @@ typedef struct {
 
 } SDL_ExtendedGameControllerBind;
 
+#if SDL_VERSION_ATLEAST(2, 26, 0)
+/* our hard coded list of mapping support */
+typedef enum {
```
Comment (SachinVin) [inline src/input_common/sdl/sdl_impl.cpp:62]: Why use typedef, instead of simply naming the enum and struct?

Comment (Steveice10) [reply src/input_common/sdl/sdl_impl.cpp:62]: This is copied directly from the original SDL structs, same as the existing SDL_ExtendedGameControllerBind above.

**Comment by Steveice10** [comment]
Works in canary but untagged so other PRs can get in.

**Comment by TheGag96** [comment]
Just built this on Artix Linux x86_64, running an Nvidia 1660 Ti. If I run `citra-qt`, I can run things just fine, but running just `citra` with a game on the command line gives me:

```
[   0.003316] Frontend <Critical> citra/emu_window/emu_window_sdl2.cpp:InitializeSDL2:120: Failed to initialize SDL2: No available video device! Exiting...
```

Which does not happen in the latest master.

**Comment by Steveice10** [comment]
> Just built this on Artix Linux x86_64, running an Nvidia 1660 Ti. If I run `citra-qt`, I can run things just fine, but running just `citra` with a game on the command line gives me:
> 
> ```
> [   0.003316] Frontend <Critical> citra/emu_window/emu_window_sdl2.cpp:InitializeSDL2:120: Failed to initialize SDL2: No available video device! Exiting...
> ```
> 
> Which does not happen in the latest master.

Could you try again with the latest PR commit?

**Comment by TheGag96** [comment]
Okay, that seems to be fixed. Though, I think I was wrong about what I said earlier - this PR still has the logging bug in it, but only when ran through `citra`. `citra-qt` I guess has the QT executable suppresses the log somehow? Not sure though since "Info"-level logging still shows up.

**Comment by FearlessTobi** [comment]
is this pr good to go?

**Comment by Steveice10** [comment]
> is this pr good to go?

Yes, all the issues have been sorted out

**Pull request merged**